### DISCUSSION TITLE
DAC installation issues [solved]

### DISCUSSION THREAD

**Discussion by Alexus-3000** [probonopd/MiniDexed#705] [Q&A] [answered]
Please help me: I connected the 1602 display with i2c, the HW-040 encoder, and the PCM5102A DAC to the Raspberry Pi Zero.

logger: Circle 46 started on Raspberry Pi Zero 512MB
logger: Revision code is 900093, compiler has been GCC 10.3.1
00:00:01.00 timer: SpeedFactor is 1.00
00:00:01.52 emmc: Found a valid version 2.00 SD card
00:00:01.52 minidexed: Compile time: Aug 5 2024 07:46:05
00:00:02.39 usbdev0-1: Device ven28e9-3 found (FS)
00:00:02.41 usbdev0-1: Product: Rantion DONNER N32
00:00:02.46 usbdev0-1: Interface int1-1-0 found
00:00:02.46 usbdev0-1: Function is not supported
00:00:02.46 usbdev0-1: Interface int1-3-0 found
00:00:02.46 usbdev0-1: Using device/interface int1-3-0
00:00:02.46 dwroot: Device configured
00:00:02.47 minidexed: Tone Generators=1, Polyphony=8
00:00:02.47 minidexed: USB In Host Node
00:00:02.47 minidexed: I2S mode
00:00:02.40 ui: LCD: HD44780 I2C
00:00:02.50 ui: LCD initialised
00:00:02.50 uibuttons: GPIO Button on pin: 11 (b)
00:00:02.50 uibuttons: Note: GPIO pin 11 is already assigned
00:00:02.50 uibuttons: Note: GPIO pin 11 is already assigned
00:00:02.50 ui: MIDI Button channel not set
00:00:02.50 ui: Button User Interface initialized
00:00:02.50 ui: Rotary encoder initialized
00:00:02.50 syxfile: Directory /sysex/voice not found
00:00:02.50 minidexed: Serial MIDI interface enabled
00:00:02.50 minidexed: Program Change: Enabled for Uoices
00:00:02.51 Performance: Number of Performance Banks: 2 (last = 2)
00:00:02.52 Performance: Loaded Default Performance Bank - Last Performance: 122
00:00:02.54 serialmididevice: Serial.Read() error: -1

After that, the system freezes

**Comment by diyelectromusic** [comment]
Can you post a photo of your hardware, especially what is wired to what on the GPIO header, and your minidexed.ini file?

The error-sounding messages you have here are pretty typical and nothing particularly unusual.

Does it do anything if you don't power up with the USB device attached?

Kevin

**Comment by Alexus-3000** [comment]
I figured out almost everything. Everything works via hdmi. There is no sound through the DAC. Everything is connected as in the sample on the website. I changed the DAC, there is no sound.
![3](https://github.com/user-attachments/assets/a43cc666-7bb7-4fc4-9e0c-9198dd4c38eb)
[minidexed.txt](https://github.com/user-attachments/files/16729794/minidexed.txt)

**Reply by diyelectromusic** [reply]
Are the solder jumpers correct on the back of the DAC?  Details here: https://github.com/dwhinham/mt32-pi/wiki/GY-PCM5102-DAC-module.

Kevin

**Comment by Alexus-3000** [comment]
I made jumpers. It hasn't worked yet, there may be soldering defects. I will continue to experiment

**Reply by probonopd** [reply]
And check the tiny solder bridges on the other side vs. what is in our wiki.

**Comment by Alexus-3000** [comment, marked as answer]
So far, I have installed the CJMCU 5102 DAC module. Works.

**Comment by Alexus-3000** [comment]
I plan to replace the encoder with buttons. If there are examples of schemes and the minindex.ini file, I would be very grateful!
